Kill declared_type and ignore_check from THFormal.

gchanan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #25737 Kill defaults in nn.yaml.
* #25717 Kill 'default_init', which isn't needed anymore.
* #25712 Kill cpu_zero in THFormal.
* **#25655 Kill declared_type and ignore_check from THFormal.**
* #25615 Kill if_true / if_false in Declarations.cwarp.
* #25614 Replace simple if_true / if_false cases in Declarations.cwrap.
* #25613 Kill aten_custom_call.
* #25612 Kill remaining defaults in Declarations.cwrap.
* #25611 Get rid of more defaults in Declarations.cwrap.
* #25610 Kill most defaults in Declarations.cwrap.

They aren't used anymore.

Differential Revision: [D17188631](https://our.internmc.facebook.com/intern/diff/D17188631)

gchanan committed:
Kill declared_type and ignore_check from THFormal.

They aren't used anymore.

zou3519 commented on aten/src/ATen/common_with_cwrap.py:
```diff
@@ -92,9 +92,9 @@ def signature(option, kwarg_only_count):
 
 
 def sort_by_number_of_options(declaration, reverse=True):
```
nit: rename this to `sort_by_number_of_args` or something similar. `options` doesn't mean anything here anymore

gchanan committed:
Update on "Kill declared_type and ignore_check from THFormal."


They aren't used anymore.

gchanan committed:
Update on "Kill declared_type and ignore_check from THFormal."


They aren't used anymore.

gchanan committed:
Update on "Kill declared_type and ignore_check from THFormal."


They aren't used anymore.

Differential Revision: [D17188631](https://our.internmc.facebook.com/intern/diff/D17188631)

gchanan committed:
Update on "Kill declared_type and ignore_check from THFormal."


They aren't used anymore.

Differential Revision: [D17188631](https://our.internmc.facebook.com/intern/diff/D17188631)

gchanan committed:
Update on "Kill declared_type and ignore_check from THFormal."


They aren't used anymore.

Differential Revision: [D17188631](https://our.internmc.facebook.com/intern/diff/D17188631)

gchanan committed:
Update on "Kill declared_type and ignore_check from THFormal."


They aren't used anymore.

Differential Revision: [D17188631](https://our.internmc.facebook.com/intern/diff/D17188631)

gchanan committed:
Update on "Kill declared_type and ignore_check from THFormal."


They aren't used anymore.

Differential Revision: [D17188631](https://our.internmc.facebook.com/intern/diff/D17188631)

gchanan committed:
Update on "Kill declared_type and ignore_check from THFormal."


They aren't used anymore.

Differential Revision: [D17188631](https://our.internmc.facebook.com/intern/diff/D17188631)

gchanan committed:
Update on "Kill declared_type and ignore_check from THFormal."


They aren't used anymore.

Differential Revision: [D17188631](https://our.internmc.facebook.com/intern/diff/D17188631)

gchanan committed:
Update on "Kill declared_type and ignore_check from THFormal."


They aren't used anymore.

Differential Revision: [D17188631](https://our.internmc.facebook.com/intern/diff/D17188631)
